uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-client/src/test/java/org/uberfire/ext/editor/commons/client/BaseEditorTest.java

```diff
@@ -369,6 +381,134 @@ public void testMakeMenuBarWhenItDoesNotContainAllMenuItems() {
         verify(menuBuilder, never()).addNewTopLevelMenu(any());
     }
 
+    @Test
+    public void testReloadWithObservablePath() {
+
+        final ObservablePath path = mock(ObservablePath.class);
+
+        doNothing().when(editor).refreshTitle(path);
+        doNothing().when(editor).showBusyIndicator();
+        doNothing().when(editor).loadContent();
+        doNothing().when(editor).notifyChangeTitle(path);
+        doNothing().when(editor).initVersionRecordManager();
+
+        editor.reload(path);
+
+        verify(editor).refreshTitle(path);
+        verify(editor).showBusyIndicator();
+        verify(editor).loadContent();
+        verify(editor).notifyChangeTitle(path);
+        verify(editor).initVersionRecordManager();
+    }
+
+    @Test
+    public void testRefreshTitleWithObservablePath() {
+
+        final ObservablePath path = mock(ObservablePath.class);
+        final String title = "title";
+
+        doReturn(title).when(editor).getTitleText(path);
+
+        editor.refreshTitle(path);
+
+        verify(baseView).refreshTitle(title);
+    }
+
+    @Test
+    public void testShowBusyIndicator() {
+
+        final String loading = "Loading...";
+
+        doReturn(loading).when(editor).makeLoading();
+
+        editor.showBusyIndicator();
+
+        verify(baseView).showBusyIndicator(loading);
+    }
+
+    @Test
+    public void testNotifyChangeTitle() {
+
+        final ObservablePath path = mock(ObservablePath.class);
+        final ChangeTitleWidgetEvent widgetEvent = mock(ChangeTitleWidgetEvent.class);
+
+        doReturn(widgetEvent).when(editor).makeChangeTitleWidgetEvent(path);
+
+        editor.notifyChangeTitle(path);
+
+        verify(changeTitleNotification).fire(widgetEvent);
+    }
+
+    @Test
+    public void testMakeChangeTitleWidgetEvent() {
+
+        final ObservablePath path = mock(ObservablePath.class);
+        final PlaceRequest placeRequest = mock(PlaceRequest.class);
+        final String title = "title";
+        final EditorTitle editorTitle = mock(EditorTitle.class);
+
+        doReturn(placeRequest).when(editor).getPlace();
+        doReturn(title).when(editor).getTitleText(path);
+        doReturn(editorTitle).when(editor).getTitleWidget();
+
+        final ChangeTitleWidgetEvent event = editor.makeChangeTitleWidgetEvent(path);
+
+        assertEquals(placeRequest, event.getPlaceRequest());
+        assertEquals(title, event.getTitle());
+        assertEquals(editorTitle, event.getTitleDecoration());
+    }
+
+    @Test
+    public void testInitVersionRecordManager() {
+
+        final ObservablePath path = mock(ObservablePath.class);
+        final PlaceRequest placeRequest = mock(PlaceRequest.class);
+        final String version = "version";
+        final Callback<VersionRecord> callback = (v) -> {
```

Comment:
Might have been more readable to simply use a mock Callback?

Reply:
Good catch. Fixed.